tests/daemon/unittest_service_db.cc

```diff
@@ -0,0 +1,280 @@
+
+  try {
+    guint version;
+    db.set_model ("", "model", true, "description", &version);
+  } catch (const std::exception &e) {
+    g_critical ("Got Exception: %s", e.what ());
+    gotException = 1;
+  }
+  EXPECT_EQ (gotException, 1);
+
+  gotException = 0;
+  try {
+    guint version;
+    db.set_model ("test", "", true, "description", &version);
+  } catch (const std::exception &e) {
+    g_critical ("Got Exception: %s", e.what ());
+    gotException = 1;
+  }
+  EXPECT_EQ (gotException, 1);
+
+  gotException = 0;
+  try {
+    guint version;
+    db.set_model ("test", "model", true, "", &version);
+  } catch (const std::exception &e) {
+    g_critical ("Got Exception: %s", e.what ());
+    gotException = 1;
+  }
+  EXPECT_EQ (gotException, 1);
+}
+
+/**
+ * @brief Negative test for get_model. Empty name, invalid version.
+ */
+TEST (serviceDB, get_model_n)
+{
+  MLServiceDB &db = MLServiceDB::getInstance ();
+  int gotException = 0;
+
+  try {
+    std::string model_description;
+    db.get_model ("", model_description, 0);
+  } catch (const std::exception &e) {
+    g_critical ("Got Exception: %s", e.what ());
+    gotException = 1;
+  }
+  EXPECT_EQ (gotException, 1);
+
+  gotException = 0;
+  try {
+    std::string model_description;
+    db.get_model ("test", model_description, -54321);
+  } catch (const std::exception &e) {
+    g_critical ("Got Exception: %s", e.what ());
+    gotException = 1;
+  }
+  EXPECT_EQ (gotException, 1);
+}
+
+/**
+ * @brief Negative test for update_model_description. Empty name or description.
+ */
+TEST (serviceDB, update_model_description_n)
+{
+  MLServiceDB &db = MLServiceDB::getInstance ();
+  int gotException = 0;
+
+  try {
+    db.update_model_description ("", 0, "description");
+  } catch (const std::exception &e) {
+    g_critical ("Got Exception: %s", e.what ());
+    gotException = 1;
+  }
+  EXPECT_EQ (gotException, 1);
+
+  gotException = 0;
+  try {
+    db.update_model_description ("test", 0, "");
+  } catch (const std::exception &e) {
+    g_critical ("Got Exception: %s", e.what ());
+    gotException = 1;
+  }
+  EXPECT_EQ (gotException, 1);
+}
+
+/**
+ * @brief Negative test for activate_model. Empty name
+ */
+TEST (serviceDB, activate_model_n)
+{
+  MLServiceDB &db = MLServiceDB::getInstance ();
+  int gotException = 0;
+
+  try {
+    db.activate_model ("", 0);
+  } catch (const std::exception &e) {
+    g_critical ("Got Exception: %s", e.what ());
+    gotException = 1;
+  }
+  EXPECT_EQ (gotException, 1);
+}
+
+/**
+ * @brief Negative test for delete_model. Empty name.
+ */
+TEST (serviceDB, delete_model_n)
+{
+  MLServiceDB &db = MLServiceDB::getInstance ();
+  int gotException = 0;
+
+  try {
+    db.delete_model ("", 0);
+  } catch (const std::exception &e) {
+    g_critical ("Got Exception: %s", e.what ());
+    gotException = 1;
+  }
+  EXPECT_EQ (gotException, 1);
+}
+
+/**
+ * @brief Negative test for model. DB is not initialized.
+ */
+TEST (serviceDB, model_n)
```

Comment:
For future release report: please separate all tests, more negative case.
(e.g. updateModelDescInvalidVer_n / setModelInvalidPath / ...)

Reply:
I added tcs for every function calls, PTAL.